File: lista_de_exercicios/questao11/questao11.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcularSalario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas_trabalhadas, valor_por_hora):
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_por_hora = valor_por_hora

    def calcularSalario(self):
        return self.horas_trabalhadas * self.valor_por_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_mensal):
        self.salario_mensal = salario_mensal

    def calcularSalario(self):
        return self.salario_mensal

File: lista_de_exercicios/questao11/questao11.go
package main

import "fmt"

type Funcionario interface {
	calcularSalario() float64
}

type FuncionarioHorista struct {
	horasTrabalhadas int
	valorPorHora     float64
}

func (f FuncionarioHorista) calcularSalario() float64 {
	return float64(f.horasTrabalhadas) * f.valorPorHora
}

type FuncionarioAssalariado struct {
	salarioMensal float64
}

func (f FuncionarioAssalariado) calcularSalario() float64 {
	return f.salarioMensal
}

func main() {
	fh := FuncionarioHorista{horasTrabalhadas: 40, valorPorHora: 15.0}
	fa := FuncionarioAssalariado{salarioMensal: 3000.0}
	fmt.Println(fh.calcularSalario())
	fmt.Println(fa.calcularSalario())
}
